es/lessons/basics/documentation.md

```diff
@@ -0,0 +1,309 @@
+
+### Documentando Módulos
+
+El anotador `@moduledoc` permite agregar documentación al nivel de módulo. Esto tipicamente se situa bajo la declaración `defmodule` al inicio del archivo. El ejemplo a continuación muestra un comentario de una linea dentro del decorador `@moduledoc`.
+
+```elixir
+defmodule Greeter do
+  @moduledoc """
+  Provides a function `hello/1` to greet a human
+  """
+
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+```
+
+Nosotros (u otros) podemos acceder a esta documentación usando la función de ayuda `h` dentro de IEx. 
+
+```elixir
+iex> c("greeter.ex")
+[Greeter]
+
+iex> h Greeter
+
+                Greeter
+
+Provides a function hello/1 to greet a human
+```
+
+### Documentando Funciones
+
+Al igual que Elixir nos da la habilidad para documentar a nivel de módulo, también habilita similares anotaciones para documentar funciones. El anotador `@doc` permite agregar documentación a nivel de función. El anotador `@doc` se situa justo sobre la función que esta comentando.
+
+```elixir
+defmodule Greeter do
+  @moduledoc """
+  ...
+  """
+
+  @doc """
+  Prints a hello message
+
+  ## Parameters
+
+    - name: String that represents the name of the person.
+
+  ## Examples
+
+      iex> Greeter.hello("Sean")
+      "Hello, Sean"
+
+      iex> Greeter.hello("pete")
+      "Hello, pete"
+
+  """
+  @spec hello(String.t) :: String.t
+  def hello(name) do
+    "Hello, " <> name
+  end
+end
+```
+
+Si entramos en IEx otra vez y usamos el comando de ayuda (`h`) con la función precedida del nombre del módulo, veremos lo siguiente:
+
+```elixir
+iex> c("greeter.ex")
+[Greeter]
+
+iex> h Greeter.hello
+
+                def hello(name)
+
+Prints a hello message
+
+Parameters
+
+  • name: String that represents the name of the person.
+
+Examples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+
+iex>
+```
+
+¿Notas como puedes usar marcado dentro de la documentación y la terminal lo renderizará? Además de ser realmente genial y una adición novedosa al vasto ecosistema de Elixir, se vuelve mucho mas interesante cuando vemos que ExDoc genera documentación en HTML sobre la marcha.
+
+**Nota:** la anotación `@spec` es usada para analisis estático del código. Para aprender mas acerca de eso, revisa la lección [Especificaciones y tipos](../../advanced/typespec).
+
+## ExDoc
+
+ExDoc es un proyecto oficial de Elixir que puede ser encontrado en [GitHub](https://github.com/elixir-lang/ex_doc). Produce **HTML (HyperText Markup Language) y documentación en linea** para proyectos Elixir. Primero vamos a crear un proyecto Mix para nuestra aplicación:
```

Comment:
It also produces epub, maybe the eng. Version could reflect this

Reply:
will there any problem when exporting to epub?

Reply:
It shouldn't, and its described as a tool for such usage:
> ExDoc produces HTML and EPUB documentation for Elixir projects.